chainhook 1.5.1 (new formula)

MicaiahReid opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,27 @@
+class Chainhook < Formula
+  desc "Reorg-aware indexing engine for the Stacks & Bitcoin blockchains"
+  homepage "https://github.com/hirosystems/chainhook"
+  # pull from git tag to get submodules
```
what is this for?

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,27 @@
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/chainhook.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
```
```suggestion
```

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,20 @@
+class Chainhook < Formula
+  desc "Reorg-aware indexing engine for the Stacks & Bitcoin blockchains"
+  homepage "https://github.com/hirosystems/chainhook"
+  url "https://github.com/hirosystems/chainhook.git",
+      tag:      "v1.4.1",
+      revision: "9d9f0d66dc32c7680fdabd3dba2ecf2847dd2b2a"
```
please use tarball

MicaiahReid replied:
Hi, thanks for the feedback @chenrui333. 

Can I ask why we need to use a tarball? I used another comparable rust formula as a template (https://github.com/Homebrew/homebrew-core/blob/master/Formula/c/clarinet.rb), and it builds from a git revision as well.

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,20 @@
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "chainhook-install", "--root", prefix.to_s
```
also use std_cargo_args in here as well, see some build change in https://github.com/Homebrew/homebrew-core/pull/168061

SMillerDev commented:
>     Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"

MicaiahReid commented:
@chenrui333 @SMillerDev I have addressed your comments!

I thought I fixed all of the commit messages to match the expected style, but it still is failing CI on that point.

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,24 @@
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "llvm@15" => :build # for libclang
+  end
```
we can do something like this?
```suggestion
  uses_from_macos "llvm" => :build # for libclang
```

MicaiahReid commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,24 @@
+  end
+
+  def install
+    system "cargo", "install", "--force", "--features", "cli", "--features",
+     "debug", "--no-default-features", *std_cargo_args(path: "components/chainhook-cli")
```
This is really just poor naming of the feature on the Chainhook side. This is how we recommend users install Chainhook.

chenrui333 replied:
any reference about that `debug` feature?

MicaiahReid commented:
@chenrui333 Could I get a re-review on this?

chenrui333 commented on Formula/c/chainhook.rb:
```diff
@@ -0,0 +1,22 @@
+  uses_from_macos "llvm" => :build # for libclang
+
+  def install
+    system "cargo", "install", "--force", "--features", "cli", "--features",
+     "debug", "--no-default-features", *std_cargo_args(path: "components/chainhook-cli")
```
```suggestion
    system "cargo", "install", "--features", "cli,debug", "--no-default-features", 
                                *std_cargo_args(path: "components/chainhook-cli")
```

chenrui333 commented:
@MicaiahReid one last comment, can you also squash all your commits as well? And add this formula into autobump.txt file. Thanks!

chenrui333 commented:
@MicaiahReid, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
